tox.ini

```diff
@@ -1,36 +1,41 @@
 [tox]
 skip_missing_interpreters = True
-envlist = py{27,35,36,37,38}-{Linux,Windows,macOS}
+envlist = py{27,35,36,37,38}-{Linux,Windows,macOS}, docs, lint
 
 [testenv]
-whitelist_externals =
-    sudo
-    make
-    cmd
 platform =
     Windows: win32
     Linux: linux
     macOS: darwin
+whitelist_externals = bash
 extras = dev
-install_command =
-    {envpython} -m pip install {opts} {packages}
-commands_pre =
-    {envpython} -m pip check
-    # Stop the test if there are Python syntax errors or undefined names.
+commands =
+    pip check
+    Linux: bash -ec "if ! getcap $(realpath {envpython}) | grep -q cap_net_admin,cap_net_raw+eip; then sudo setcap cap_net_admin,cap_net_raw+eip $(realpath {envpython}); fi"
```

Comment:
This line is new, right? What's the gist of it?

Reply:
From the PR comment:

> Only exception is a one-line bash script on Linux, which sets `cap_net_admin` and `cap_net_raw+eip` on the tox python interpreter. This is done every time the tox env is (re-)created and requires root privileges. Subsequent runs won't need any special privileges.
With this change we don't run pytest with sudo, so there won't be any cache/log/output/data files owned by root in your project folder. On top it's more secure, as we only grant the capabilities that are really required.

It checks if the required capabilities needed for raw socket operations are set with `getcap` and sets them with `setcap` if they aren't already set.
The capabilities are only applied to the tox env interpreter, not the global one.

Reply:
Ah that makes sense now. I like the security hygiene!